packages/starlight/schemas/sidebar.ts

```diff
@@ -0,0 +1,89 @@
+import type { AstroBuiltinAttributes } from 'astro';
+import type { HTMLAttributes } from 'astro/types';
+import { z } from 'astro/zod';
+import { BadgeConfigSchema } from './badge';
+
+const SidebarBaseSchema = z.object({
+	/** The visible label for this item in the sidebar. */
+	label: z.string(),
+	/** Translations of the `label` for each supported language. */
+	translations: z.record(z.string()).default({}),
+});
+
+const SidebarGroupSchema = SidebarBaseSchema.extend({
+	/** Whether this item should be collapsed by default. */
+	collapsed: z.boolean().default(false),
+});
+
+// HTML attributes that can be added to an anchor element, validated as
+// `Record<string, string | number | boolean | undefined>` but typed as `HTMLAttributes<'a'>`
+// for user convenience.
+const linkHTMLAttributesSchema = z.record(
+	z.union([z.string(), z.number(), z.boolean(), z.undefined()])
+) as z.Schema<Omit<HTMLAttributes<'a'>, keyof AstroBuiltinAttributes | 'children'>>;
+export type LinkHTMLAttributes = z.infer<typeof linkHTMLAttributesSchema>;
+
+export const SidebarLinkItemHTMLAttributesSchema = () => linkHTMLAttributesSchema.optional();
```

Comment:
Would a default value here also make sense? Would avoid us needing to do any optional chaining when reading from it later and I don’t think we’re relying on this being undefined?

```suggestion
export const SidebarLinkItemHTMLAttributesSchema = () => linkHTMLAttributesSchema.default({});
```

Reply:
I think so, I actually started to implement it with a default value at first and I refactored to an optional value later on just to match the badges implementation but it's not really necessary.